diff --git a/softkeymaster/module.cpp b/softkeymaster/module.cpp
index bfb1dd5..d14b5ff 100644
--- a/softkeymaster/module.cpp
+++ b/softkeymaster/module.cpp
@@ -58,7 +58,7 @@ static int openssl_open(const hw_module_t* module, const char* name, hw_device_t
     dev->common.module = (struct hw_module_t*)module;
     dev->common.close = openssl_close;
 
-    dev->flags = KEYMASTER_SOFTWARE_ONLY;
+    dev->flags = 0;//KEYMASTER_SOFTWARE_ONLY;
 
     dev->generate_keypair = openssl_generate_keypair;
     dev->import_keypair = openssl_import_keypair;
